src/js/components/FrameworkConfiguration.js

```diff
@@ -0,0 +1,387 @@
+import deepEqual from "deep-equal";
+
+import FullScreenModal from "#SRC/js/components/modals/FullScreenModal";
+import FullScreenModalHeader
+  from "#SRC/js/components/modals/FullScreenModalHeader";
+import FullScreenModalHeaderActions
+  from "#SRC/js/components/modals/FullScreenModalHeaderActions";
+import FullScreenModalHeaderTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderTitle";
+import FullScreenModalHeaderSubTitle
+  from "#SRC/js/components/modals/FullScreenModalHeaderSubTitle";
+import ToggleButton from "#SRC/js/components/ToggleButton";
+import HashMapDisplay from "#SRC/js/components/HashMapDisplay";
+import ModalHeading from "#SRC/js/components/modals/ModalHeading";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import Icon from "#SRC/js/components/Icon";
+import Util from "#SRC/js/utils/Util";
+import StringUtil from "#SRC/js/utils/StringUtil";
+import CosmosErrorMessage from "#SRC/js/components/CosmosErrorMessage";
+import FrameworkConfigurationForm
+  from "#SRC/js/components/FrameworkConfigurationForm";
+
+export default class FrameworkConfiguration extends Component {
+  constructor(props) {
+    super(props);
+
+    const schema = props.packageDetails.getConfig();
+    const activeTab = Object.keys(schema.properties)[0];
+    const focusField = Object.keys(schema.properties[activeTab].properties)[0];
+
+    this.state = {
+      reviewActive: props.isInitialDeploy,
+      activeTab,
+      focusField,
+      jsonEditorActive: false,
+      hasChangesApplied: false,
+      isConfirmOpen: false,
+      isOpen: true
+    };
+  }
+
+  onFormDataChange(formData) {
+    this.props.onFormDataChange(formData);
+
+    this.setState({ hasChangesApplied: true });
+  }
+
+  onFormErrorChange(formErrors) {
+    this.props.onFormErrorChange(formErrors);
+  }
+
+  onFocusFieldChange(activeTab, focusField) {
+    if (deepEqual(focusField, this.state.focusField)) {
+      return false;
+    }
+
+    this.setState({ focusField, activeTab });
+  }
+
+  onActiveTabChange(activeTab) {
+    const { currentActiveTab } = this.state;
+    const { packageDetails } = this.props;
+
+    if (deepEqual(activeTab, currentActiveTab)) {
+      return false;
+    }
+
+    const schema = packageDetails.getConfig();
+    const focusField = Object.keys(schema.properties[activeTab].properties)[0];
+
+    this.setState({ activeTab, focusField });
+  }
+
+  onReviewConfigurationRowClick(rowData) {
+    const { tabViewID } = rowData;
+
+    const keyPath = (tabViewID || "").split(".");
+    if (keyPath.length < 2) {
+      this.onEditConfigurationButtonClick();
+
+      return false;
+    }
+
+    const activeTab = keyPath[0];
+    const focusField = keyPath[1];
+
+    this.setState({ reviewActive: false, focusField, activeTab });
+  }
+
+  onEditConfigurationButtonClick() {
+    const { packageDetails } = this.props;
+
+    const schema = packageDetails.getConfig();
+    const activeTab = Object.keys(schema.properties)[0];
```

Comment:
Another option to get the first element would be:
```JS
 const [activeTab] = Object.keys(schema.properties);
```
But this is personal taste. As it is not adding a lot of readability feel free to adjust in this and next line.